sentry/src/routes/channel.rs

```diff
@@ -392,6 +398,95 @@ pub async fn get_accounting_for_channel<C: Locked + 'static>(
     Ok(success_response(serde_json::to_string(&res)?))
 }
 
+pub async fn channel_payout<C: Locked + 'static>(
+    req: Request<Body>,
+    app: &Application<C>,
+) -> Result<Response<Body>, ResponseError> {
+    let channel_context = req
+        .extensions()
+        .get::<ChainOf<Channel>>()
+        .expect("Request should have Channel & Chain/TokenInfo")
+        .to_owned();
+
+    let body = hyper::body::to_bytes(req.into_body()).await?;
+    let to_pay = serde_json::from_slice::<ChannelPayRequest>(&body)
+        .map_err(|e| ResponseError::FailedValidation(e.to_string()))?;
+
+    let channel_campaigns = fetch_campaign_ids_for_channel(
+        &app.pool,
+        channel_context.context.id(),
+        app.config.campaigns_find_limit,
+    )
+    .await?;
+
+    let campaigns_remaining_sum = app
+        .campaign_remaining
+        .get_multiple(channel_campaigns.as_slice())
```

Comment:
FYI: You don't usually need `as_slice()`, rust will automagically do this when you use `&` for a `Vec<T>` when you need `&[T]`

Do check out the `Deref` trait